pkg/test/kube/util.go

```diff
@@ -340,3 +341,43 @@ func checkAllNamesExist(names []string, haystack []string) bool {
 
 	return true
 }
+
+// Resolve domain name and return ip address.
+// By default, return ipv4 address and if missing, return ipv6.
+func resolveHostDomainToIP(hostDomain string) (string, error) {
+	ips, err := net.LookupIP(hostDomain)
+	if err != nil {
+		return "", err
+	}
+
+	var ipv6Addr string
+
+	for _, ip := range ips {
+		if ip.To4() != nil {
+			return ip.String(), nil
+		} else if ipv6Addr == "" {
+			ipv6Addr = ip.String()
+		}
+	}
+
+	if ipv6Addr != "" {
+		return ipv6Addr, nil
+	}
+
+	return "", fmt.Errorf("no IP address found for hostname: %s", hostDomain)
+}
+
+// When the Ingress is a domain name (in public cloud), it might take a bit time to make it reachable.
+func WaitUntilReachableIngress(hostDomain string) (string, error) {
+	var ip string
+	err := retry.UntilSuccess(func() error {
+		ipAddr, err := resolveHostDomainToIP(hostDomain)
+		if err != nil {
+			return err
+		}
+		ip = ipAddr
+		return nil
+	}, retry.Timeout(90*time.Second), retry.BackoffDelay(1*time.Second))
```

Comment:
Curious why the timeout was updated to 90 sec from a minute. Not opposed, just curious :)

Reply:
Hi @jaellio 

With the timeout of a minute, the check was inconsistent and failed from time to time.
Since I can't predict the time it could take for the domain to become resolvable and it fully depends on the DNS servers, I decided to increase the timeout to make sure it will not fail ahead of time.